src/codex_plus/llm_execution_middleware.py

```
@@ -365,71 +455,698 @@ async def process_request(self, request, path: str):
+                                            yield args_event
+                                            sequence_number += 1
+                                            function_arguments_chunks.append(arguments)
+
+                                if finish_reason:
+                                    tool_call_result_text: Optional[str] = None
+                                    if finish_reason == "tool_calls" and function_arguments_chunks:
+                                        arguments_json = "".join(function_arguments_chunks)
+                                        try:
+                                            call_payload = json.loads(arguments_json)
+                                        except json.JSONDecodeError as exc:
+                                            logger.error(f"Failed to parse tool arguments: {exc}")
```

Ruff (0.13.3)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)